hollow/src/main/java/com/netflix/hollow/api/client/HollowClientUpdater.java

```diff
@@ -66,18 +66,22 @@ public HollowClientUpdater(HollowConsumer.BlobRetriever transitionCreator,
                                HollowMetricsCollector<HollowConsumerMetrics> metricsCollector) {
         this.planner = new HollowUpdatePlanner(transitionCreator, doubleSnapshotConfig);
         this.failedTransitionTracker = new FailedTransitionTracker();
-        this.staleReferenceDetector = new StaleHollowReferenceDetector(objectLongevityConfig, objectLongevityDetector);
+        this.staleReferenceDetector = objectLongevityConfig != null && objectLongevityDetector != null
```

Comment:
I usually prefer to reduce the number of null checks in code. We already have `ObjectLongevityConfig.DEFAULT_CONFIG`. I could introduce a `NULL_CONFIG` or `NOOP_CONFIG`.

However, one thing I like about `null` here is it signals clear intent to permanently disable the feature, so we can leave out the detector (and its monitor thread) entirely.

Reply:
👍

Reply:
`null` checks are cheap (HotSpot optimizes these away very efficiently), so if performance was a concern it need not be.

What if only one of the values is `null`. Then a `StaleHollowReferenceDetector` instance will be created and likely result in an NPE. You might wanna trigger this off config and if non-`null` requiring `objectLongevityDetector` to be non-`null` i.e. there should be `null` checks in `StaleHollowReferenceDetector` constructor for both values.